### DISCUSSION TITLE
Transmission Not Starting - Missing Config Files

### DISCUSSION THREAD

**Discussion by harkus** [haugene/docker-transmission-openvpn#2445] [General]
Trying to restart Transmission via docker and get the following error before quitting:
Using OpenVPN provider: VPNUNLIMITED
Running with VPN_CONFIG_SOURCE auto
No bundled config script found for VPNUNLIMITED. Defaulting to external config
**Downloading configs from https://github.com/haugene/vpn-configs-contrib/archive/main.zip into /tmp/tmp.kqj0uEgfXX
curl: (6) getaddrinfo() thread failed to start
./etc/openvpn/fetch-external-configs.sh: line 18: config_repo_temp_dir: unbound variable**

It appears that https://github.com/haugene/vpn-configs-contrib/archive no longer exists providing a 404 error when I try to access it. So, obviously, the main.zip file is not accessible.  - scratch this, it's accessible.

Any idea of a fix or how to work around this by using local config files for my VPN?

**Comment by ihme1492** [comment]
I have been having the same issue for a few days now myself.

**Comment by pow3rslave** [comment]
i've had something similar occurring since i updated to latest version

> Transmission    | Starting container with revision: b33d0fe4c938259a0d4eb844e55468f387456121
> Transmission    | Creating TUN device /dev/net/tun
> Transmission    | Using OpenVPN provider: PIA
> Transmission    | Running with VPN_CONFIG_SOURCE auto
> Transmission    | No bundled config script found for PIA. Defaulting to external config
> Transmission    | Downloading configs from https://github.com/haugene/vpn-configs-contrib/archive/main.zip into /tmp/tmp.wwq4AM6VJJ
> Transmission    | Extracting configs to /tmp/tmp.ttFhYw3kDZ
> Transmission    | Found configs for PIA in /tmp/tmp.ttFhYw3kDZ/vpn-configs-contrib-main/openvpn/pia, will replace current content in /etc/openvpn/pia
> Transmission    | Cleanup: deleting /tmp/tmp.wwq4AM6VJJ and /tmp/tmp.ttFhYw3kDZ
> Transmission    | Supplied config au_melbourne.ovpn could not be found.
> Transmission    | Your options for this provider are:

looking at the PIA directory under  https://github.com/haugene/vpn-configs-contrib/archive/main.zip , it doesn't contain any .ovpn files, instead it looks like it tries to grab a specific bundle, so this may be where it's failing.

as a workaround, you can set your VPN provider up as custom, by following the steps for  **single ovpn provider** here
https://haugene.github.io/docker-transmission-openvpn/supported-providers/

**Reply by harkus** [reply]
I see that PIA is missing .opvn files but VPNUnlimited has the ones I need in the folder within the zip file.

It was working until around midday yesterday. I hadn't even done an update until AFTER this was occurring, though I don't think the update happened for Transmission. I had restart transmission a few times before then.

I'm very curious why it's not working though when everything looks correct. So frustrating.

**Reply by harkus** [reply]
I downloaded my own OVPN file and mapped it and configured transmission but it gave other errors

Starting container with revision: b33d0fe4c938259a0d4eb844e55468f387456121
/etc/openvpn/start.sh: line 34:    11 Aborted                 (core dumped) nslookup ${HEALTH_CHECK_HOST:-"google.com"} > /dev/null 2>&1
WARNING: initial DNS resolution test failed
Creating TUN device /dev/net/tun
Using OpenVPN provider: CUSTOM
Modifying /etc/openvpn/custom/default.ovpn for best behaviour in this container
Modification: Point auth-user-pass option to the username/password file
Modification: Change ca certificate path
Modification: Change ping options
Modification: Update/set resolv-retry to 15 seconds
Modification: Change tls-crypt keyfile path
Modification: Set output verbosity to 3
Modification: Remap SIGUSR1 signal to SIGTERM, avoid OpenVPN restart loop
Modification: Updating status for config failure detection
Setting OpenVPN credentials...
adding route to local network 192.168.0.0/24 via 172.21.0.1 dev eth0
2022-11-25 13:02:19 DEPRECATED OPTION: --cipher set to 'AES-256-CBC' but missing in --data-ciphers (AES-256-GCM:AES-128-GCM). Future OpenVPN version will ignore --cipher for cipher negotiations. Add 'AES-256-CBC' to --data-ciphers or change --cipher 'AES-256-CBC' to --data-ciphers-fallback 'AES-256-CBC' to silence this warning.
2022-11-25 13:02:19 OpenVPN 2.5.5 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [PKCS11] [MH/PKTINFO] [AEAD] built on Jul 14 2022
2022-11-25 13:02:19 library versions: OpenSSL 3.0.2 15 Mar 2022, LZO 2.10
2022-11-25 13:02:19 NOTE: the current --script-security setting may allow this configuration to call user-defined scripts
2022-11-25 13:02:19 OpenSSL: error:0A00018E:SSL routines::ca md too weak
2022-11-25 13:02:19 Cannot load inline certificate file
2022-11-25 13:02:19 Exiting due to fatal error


Just doesn't make sense.

**Reply by harkus** [reply]
Looking at my previous message:


Starting container with revision: ...
WARNING: initial DNS resolution test failed
/etc/openvpn/start.sh: line 34:    10 Aborted                 (core dumped) nslookup ${HEALTH_CHECK_HOST:-"google.com"} > /dev/null 2>&1
Creating TUN device /dev/net/tun
Using OpenVPN provider: VPNUNLIMITED
Running with VPN_CONFIG_SOURCE auto
No bundled config script found for VPNUNLIMITED. Defaulting to external config
Downloading configs from https://github.com/haugene/vpn-configs-contrib/archive/main.zip into /tmp/tmp.cjKoxFPCH7
curl: (6) getaddrinfo() thread failed to start
./etc/openvpn/fetch-external-configs.sh: line 18: config_repo_temp_dir: unbound variable


It looks like the container is having issues accessing google.com, maybe that's the problem. Any idea how to fix the DNS?

Here's my Docker-compose file:
  transmission:
    container_name: transmission
    cap_add:
      - NET_ADMIN
    volumes:
      - /mnt/drive/docker-config/downloads:/downloads
      - /mnt/drive/docker-config:/data
    environment:
      - OPENVPN_PROVIDER=VPNUNLIMITED
      - OPENVPN_CONFIG=ca-vn
      - OPENVPN_USERNAME=user
      - OPENVPN_PASSWORD=pass
      - LOCAL_NETWORK=192.168.0.0/24
      - TZ=${TZ}
      - PUID=${PUID}
      - PGID=${PGID}
      - TRANSMISSION_WEB_UI=transmission-web-control
      - CREATE_TUN_DEVICE=true
      - TRANSMISSION_DOWNLOAD_DIR=/downloads/complete
      - TRANSMISSION_INCOMPLETE_DIR=/downloads/incomplete
      - TRANSMISSION_WATCH_DIR=/downloads/watch
    logging:
      driver: json-file
      options:
        max-size: 10m
    ports:
      - '9092:9091'
    image: haugene/transmission-openvpn:latest
    dns:
      - 8.8.8.8
      - 8.8.4.4

I added the DNS lines because of that issue.

**Comment by harkus** [comment]
Okay, good news, I got it running again. Bad news, I needed to revert to 4.2 and I lost all my in-progress torrents.
Due to a change in 4.2, I changed my /data volume to /config and everything came back and it started working perfectly again. Wohoo!
Now to fix the newest version so we can update without issue.
